plugin.video.srgssr_ch_replay/addon.py

```diff
@@ -0,0 +1,387 @@
+        xbmcgui.Dialog().ok(tr(30006), str(e.__class__.__name__), str(e))
+    return response
+
+
+def choose_tv_show_letter(channel):
+    nextMode = 'listTvShows'
+    _add_letter(channel, '#', tr(30019), nextMode)
+    for c in ascii_lowercase:
+        _add_letter(channel, c, c, nextMode)
+    xbmcplugin.endOfDirectory(handle=pluginhandle, succeeded=True)
+
+
+def _add_letter(channel, letter, letterDescription, mode):
+    directoryurl = sys.argv[0] + "?mode=" + str(mode) + "&channel=" + str(channel) + "&letter=" + letter
+    liz = xbmcgui.ListItem(letterDescription)
+    return xbmcplugin.addDirectoryItem(pluginhandle, url=directoryurl, listitem=liz, isFolder=True)
+
+
+def _add_show(name, url, mode, desc, iconimage, channel, numberOfEpisodes):
+    """
+    helper method to create a folder with subitems
+    """
+    directoryurl = sys.argv[0] + "?url=" + urllib.parse.quote_plus(url) + "&mode=" + str(mode) + "&showbackground=" + urllib.parse.quote_plus(iconimage) + \
+        "&channel=" + str(channel) + "&numberOfEpisodes=" + str(numberOfEpisodes or "")
+    liz = xbmcgui.ListItem(name)
+    liz.setLabel2(desc)
+    liz.setArt({'poster': iconimage, 'banner': iconimage, 'fanart': iconimage, 'thumb': iconimage})
+    liz.setInfo(type="Video", infoLabels={"title": name, "plot": desc, "plotoutline": desc})
+    xbmcplugin.setContent(pluginhandle, 'tvshows')
+    ok = xbmcplugin.addDirectoryItem(pluginhandle, url=directoryurl, listitem=liz, isFolder=True)
+    return ok
+
+
+def _addLink(name, url, mode, desc, iconurl, length, pubdate, showbackground, channel):
+    """
+    helper method to create an item in the list
+    """
+    linkurl = sys.argv[0] + "?url=" + urllib.parse.quote_plus(url) + "&mode=" + str(mode) + "&channel=" + str(channel)
+    liz = xbmcgui.ListItem(name)
+    liz.setLabel2(desc)
+    liz.setArt({'poster': iconurl, 'banner': iconurl, 'fanart': showbackground, 'thumb': iconurl})
+    liz.setInfo(type='Video', infoLabels={"Title": name, "Duration": length, "Plot": desc, "Aired": pubdate})
+    liz.setProperty('IsPlayable', 'true')
+    xbmcplugin.setContent(pluginhandle, 'episodes')
+    ok = xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]), url=linkurl, listitem=liz)
+    return ok
+
+
+def _addnextpage(name, url, mode, desc, showbackground, pageNumber, channel, numberOfEpisodes, nextParam):
+    """
+    helper method to create a folder with subitems
+    """
+    directoryurl = sys.argv[0] + "?url=" + urllib.parse.quote_plus(url) + "&mode=" + str(mode) + "&showbackground=" + urllib.parse.quote_plus(showbackground) + \
+        "&page=" + str(pageNumber or "") + "&channel=" + str(channel) + "&numberOfEpisodes=" + str(numberOfEpisodes or "") + "&next=" + str(nextParam)
+    liz = xbmcgui.ListItem(name)
+    liz.setLabel2(desc)
+    #liz.setArt({'poster' : '' , 'banner' : '', 'fanart' : showbackground, 'thumb' : ''})
+    liz.setInfo(type="Video", infoLabels={"title": name, "plot": desc, "plotoutline": desc})
```

Comment:
Setinfo is deprecated in nexus jfyi

Reply:
@enen92 thank you, we will have a look at it and change it with the next pull request

Reply:
Note two different code paths are required though, as the new one (using infotag member functions) is not compatible with matrix. SetInfo will probably be removed by v21